### PR TITLE
feat(webhooks/github): Support for GitHub branch events

### PR DESCRIPTION
**Pull request by jervi**
Lets you use branch creation or deletion events from GitHub as triggers. Example delete event:
```json
"trigger": {
    "type": "git",
    "user": "anonymous",
    "parameters": {},
    "artifacts": [],
    "notifications": [],
    "rebake": false,
    "dryRun": false,
    "strategy": false,
    "hash": "",
    "source": "github",
    "project": "mygithuborg",
    "branch": "my_new_branch",
    "slug": "jervi-is-testing-echo",
    "action": "branch:delete",
    "resolvedExpectedArtifacts": [],
    "expectedArtifacts": [],
    "eventId": "9ca8443c-efe2-4ca6-b1a3-5c018651b579",
    "secret": "mysecret",
    "enabled": true,
    "id": "320732db-9c80-30bf-96d8-c17383e953ff",
    "preferred": false,
    "properties": {}
},
```

### CODE REVIEW

File: echo-webhooks/src/test/java/com/netflix/spinnaker/echo/scm/BitbucketWebhookEventHandlerTest.java [hunk 1]
```diff
@@ -51,6 +52,7 @@ void canHandlePayload() throws IOException {
     event.content.put("event_type", "repo:push");
 
     BitbucketWebhookEventHandler handler = new BitbucketWebhookEventHandler();
-    assertThatCode(() -> handler.handle(event, payload)).doesNotThrowAnyException();
```
Comment (mattgogerly) [inline echo-webhooks/src/test/java/com/netflix/spinnaker/echo/scm/BitbucketWebhookEventHandlerTest.java:54]: Any way we could get a slightly more rigorous test for this?

Comment (jervi) [reply echo-webhooks/src/test/java/com/netflix/spinnaker/echo/scm/BitbucketWebhookEventHandlerTest.java:54]: Done 👍

**Comment by jervi** [comment]
@mattgogerly Can you please take another look at this?

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/controllers/WebhooksController.groovy [hunk 2]
```diff
@@ -73,6 +75,7 @@ class WebhooksController {
     if (headers.containsKey('X-Event-Key')) {
       event.content.event_type = headers['X-Event-Key'][0]
     }
+    def filteredHeaders = CollectionUtils.toMultiValueMap(headers.findAll { it.key.startsWith("x-github") })
```
Comment (mattgogerly) [inline echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/controllers/WebhooksController.groovy:78]: Does casing matter on the header? The headers above are cased, and [this doc](https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api#hello-world) has a mixture.

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/github/GithubBranchEvent.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 Netflix, Inc.
```
Comment (mattgogerly) [inline echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/github/GithubBranchEvent.java:2]: You can change this :)

File: echo-webhooks/src/test/java/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
+package com.netflix.spinnaker.echo.scm;
+
```
Comment (mattgogerly) [inline echo-webhooks/src/test/java/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandlerTest.java:2]: No copyright on this file - @dbyron-sf is that required?

Comment (jervi) [reply echo-webhooks/src/test/java/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandlerTest.java:2]: I can add one

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/controllers/WebhooksController.groovy (hunk 2 above)
Comment (jervi) [reply echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/controllers/WebhooksController.groovy:78]: When I tested it I received lowercased headers. But good point, I can make the check ignore casing.

**Review by mattgogerly — APPROVED** [review]

**Comment by mattgogerly** [comment]
@Mergifyio update

**Pull request merged**